internal/validitywindow/dependencies.go

```diff
@@ -16,9 +16,9 @@ type ExecutionBlock[Container emap.Item] interface {
 	Timestamp() int64
 	Height() uint64
 	Txs() []Container
-	ContainsTx(ids.ID) bool
+	Contains(ids.ID) bool
 }
 
 type ChainIndex[Container emap.Item] interface {
-	GetExecutionBlock(ctx context.Context, blkID ids.ID) (ExecutionBlock[Container], error)
+	GetExecutionBlock(ctx context.Context, blkID ids.ID) (ExecutionBlock[Container], bool, error)
```

Comment:
Why are we adding the boolean here?

Reply:
My guess is that this is to return `true/false`, and use the `error` if something goes wrong instead of just returning the block type + an error, and you have to introspect the error using a check against `database.ErrNotFound`.

Reply:
@aaronbuchwald see @joshua-kim request above to change the function signature to `GetExecutionBlock() (ExecutionBlock[T], bool, error) `

Reply:
When is it valid to not have an execution block that the validity window requests?

We should always have a window of blocks that goes past the time validity window. I am not sure of any case where not having a block ie. returning false should not also be treated as an error. Unless there is such a case, I think we're better off just returning an error and not adding to the return type here.